src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -48,25 +49,29 @@
   @Autowired private ProductReference productReference;
 
   @Override
-  public List<CaseDTO> getHHCaseByUPRN(final UniquePropertyReferenceNumber uprn)
+  public CaseDTO getLatestValidNonHICaseByUPRN(final UniquePropertyReferenceNumber uprn)
       throws CTPException {
 
     String uprnValue = Long.toString(uprn.getValue());
     log.with("uprn", uprn).debug("Fetching case details by UPRN");
 
     List<CollectionCase> rmCase = dataRepo.readCollectionCasesByUprn(uprnValue);
-    List<CollectionCase> results =
+    Optional<CollectionCase> result =
         rmCase
             .stream()
-            .filter(c -> c.getCaseType().equals(CaseType.HH.name()))
-            .collect(Collectors.toList());
-    List<CaseDTO> caseData = mapperFacade.mapAsList(results, CaseDTO.class);
-
-    log.with("cases", caseData.size())
-        .with("uprn", uprnValue)
-        .debug("HH case(s) retrieved for UPRN");
-
-    return caseData;
+            .filter(c -> !c.getCaseType().equals(CaseType.HI.name()))
```

Comment:
Why the the change from equals HH to not-equals HI ? 
I'm sure you have a good reason but I just can't see it

Reply:
Added comment to README file